packages/smithy-http/src/smithy_http/aio/interfaces/__init__.py

```diff
@@ -12,14 +12,23 @@
     HTTPRequestConfiguration,
 )
 
+Config = TypeVar("Config", contravariant=True)
+
+
+class EndpointParametersProtocol(Protocol[Config]):
+    @classmethod
+    def build(cls, config: Config) -> Self:
+        raise NotImplementedError()
+
+
 # EndpointParams are defined in the generated client, so we use a TypeVar here.
 # More specific EndpointParams implementations are subtypes of less specific ones. But
 # consumers of less specific EndpointParams implementations are subtypes of consumers
 # of more specific ones.
 EndpointParams = TypeVar("EndpointParams", contravariant=True)
 
 
-class EndpointResolver(Protocol[EndpointParams]):
+class EndpointResolverProtocol(Protocol[EndpointParams]):
```

Comment:
Adding type information in the name of the type isn't a thing we should be doing. Especially in Python, where you can just alias the name on import to avoid conflicts.